enhancements/gep/001-loadbalancer/README.md

```diff
@@ -0,0 +1,118 @@
+        - [Story 1](#story-1)
+        - [Story 2](#story-2)
+    - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+    - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+    - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+- [Implementation History](#implementation-history)
+\
+<!-- /toc -->
+
+## Summary
+The main goal of this proposal is to replace dns balancing with a software loadbalancer.
+Thus, all requests to the kube-apiserver from Kubernetes components (kubelet, scheduler, control-manager, kube-proxy)
+will go through this loadbalancer. Also, requests to the docker registry must also go through the loadbalancer.
+
+## Motivation
+At the moment, Gravity is using CoreDNS to determine which master to send requests to,
+but DNS is not a good tool for balancing traffic.
+Implementation of this proposal will provide the following improvements
+  1. will evenly distribute the load on the apiserver kubernetes
+  1. Improve the stability and scalability of the cluster
+  1. nodes will react faster on the disaster of the masters and recovery process
+  1. the update will be faster and more efficient
+
+### Goals
+When creating a cluster, the user can choose which load balancer will be external or internal.
+External loadbalancer is not controlled or configured by Gravity, it is the user's responsibility.
+The internal loadbalancer, on the contrary, is controlled and configured by Gravity, the planet-agent is responsible for this.
+When using an internal loadbalancer, a loadbalancer (HAProxy) will be installed for each node in the cluster including the master.
+When changing the configuration of the masters, the agent should automatically change the loadbalancer configuration on each node.
+All Kubernetes components (kubelet, scheduler, control-manager, kube-proxy) will use the loadbalancer to access the apiserver.
+
+### Non-Goals
+
+## Proposal
+Change DNS balancer to a software loadbalancer (HAProxy).
+CoreDNS will resolve the `leader.telekube.local` address, which will point either to the external address (domain name or IP) of the loadbalancer or to 127.0.0.1 when using the internal loadbalancer address.
+Image names should remain unchanged and start with `leader.telekube.local:5000`.
+To do this, the docker register must use port 5001, and port 5000 must be open on the loadbalancer.
+
+### User Stories (Optional)
+#### Story 1
+As a cluster administrator, I can set the loadbalancer settings when creating a cluster. Set the type of loadbalancer external or internal and indicate the external address.
+#### Story 2
+As a cluster administrator, I can change the loadbalancer settings on a running cluster
+
+### Risks and Mitigations
+The `leader.telekube.local` address will not point to the master node.
```

Comment:
I thought it was clear if they are using leader.telekube.local to connect to the kubernetes api then it would be better to use 127.0.0.1. If for some incomprehensible purpose, then it is better to find another solution